src/common/ipc/client_router.cpp

```diff
@@ -125,12 +125,12 @@ class ClientRouterImpl final : public ClientRouter
         GatewayImpl& operator=(const GatewayImpl&)     = delete;
         GatewayImpl& operator=(GatewayImpl&&) noexcept = delete;
 
-        ~GatewayImpl()
+        virtual ~GatewayImpl()
```

Comment:
Older clang-tidy gives here false positive, I believe.
Anyway, I decided to silence it by agreeing to make it `virtual` - not a big deal.